src/ClientWidgetApi.ts

```diff
@@ -824,6 +828,28 @@ export class ClientWidgetApi extends EventEmitter {
         }
     }
 
+    private async handleDownloadFile(request: IDownloadFileActionFromWidgetActionRequest) {
+        if (!this.hasCapability(MatrixCapabilities.MSC4039DownloadFile)) {
+            return this.transport.reply<IWidgetApiErrorResponseData>(request, {
+                error: { message: "Missing capability" },
+            });
+        }
+
+        try {
+            const result = await this.driver.downloadFile(request.data.content_uri);
+
+            return this.transport.reply<IDownloadFileActionFromWidgetResponseData>(
+                request,
+                { file: result.file },
```

Comment:
Again, more for the MSC but... what about large files? I guess the answer is that since there's no specific limit on postMessage, we can add a different API later if it's needed?

Reply:
I don't see an obvious issue with postMessage at a glance. Mostly, we assume that we can do whatever we already can do for uploading. This update to MSC4039 seeks to address Authenticated Media, which got overlooked so far for the widget API. The real answer is probably to revamp the widget API entirely to enable widgets to use their own scoped access tokens to access the C-S API directly, but we won't get there before Matrix 1.12.

Reply:
Well, I don't think trying to download a 1GB file over the postmessage API is going to be a great user experience, but yeah, if this is more of a stop-gap solution until we have scoped access tokens then that's probably okay.